lib/src/hooks/useDynamicForm.tsx

```diff
@@ -91,73 +94,83 @@ type OnChangeFormItem = (item: DynamicFormItem) => OnChangeInput;
 type OnChangeInput = (e: InputChange) => void;
 type ShouldRenderFormItem = (sectionKey: string, itemKey: string) => boolean;
 
-export function useDynamicForm(
-    sections: DynamicFormSection[]
-): [() => JSX.Element, any, React.Dispatch<React.SetStateAction<DynamicFormInternalState>>, React.Dispatch<React.SetStateAction<DynamicFormErrors>>, DynamicFormMetaData, boolean] {
-    const initialState: DynamicFormInternalState = {};
-    sections?.map((section) => {
-        initialState[section?.key] = {};
-        section.items?.map((item) => {
-            const { key, value, multi, controlType }: DynamicFormItem = item;
-            let initialValue: any;
+export type FormRenderFunction = () => JSX.Element;
+export type SetDynamicFormState = React.Dispatch<React.SetStateAction<DynamicFormInternalState>>;
+export type SetDynamicFormErrors = React.Dispatch<React.SetStateAction<DynamicFormErrors>>;
+export type SetDynamicFormWarnings = React.Dispatch<React.SetStateAction<DynamicFormErrors>>;
+type isDynamicFormDirty = boolean;
+export type UseDynamicForm = [FormRenderFunction, DynamicFormInternalState, SetDynamicFormState, SetDynamicFormErrors, SetDynamicFormWarnings, DynamicFormMetaData, isDynamicFormDirty];
+export function useDynamicForm(sections: DynamicFormSection[]): UseDynamicForm {
+    const initialState: DynamicFormInternalState = useMemo(() => {
+        let initialFormState: DynamicFormInternalState = {};
```

Comment:
notice there's no reference changes, perhaps we can use `const initialFormState`